pkg/engine/datasource/introspection_datasource/source.go

```diff
@@ -0,0 +1,88 @@
+package introspection_datasource
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+
+	"github.com/jensneuse/graphql-go-tools/pkg/introspection"
+)
+
+type Source struct {
+	introspectionData *introspection.Data
+}
+
+func (s *Source) Load(ctx context.Context, input []byte, w io.Writer) (err error) {
+	// fix corrupted input json
+	// TODO: remove when default argument value bug will be fixed
+	input = bytes.Replace(input, []byte(`"include_deprecated":}`), []byte(`"include_deprecated":false}`), 1)
+
+	var req introspectionInput
+	if err := json.Unmarshal(input, &req); err != nil {
+		return err
+	}
+
+	switch req.RequestType {
```

Comment:
What happens if a request contains multiple root fields with different Queries? e.g. requesting two type fields or type fields and schema?